gamemodes/terrortown/entities/weapons/weapon_ttt_spawneditor.lua

```diff
@@ -0,0 +1,530 @@
+				local spawn = spawns[i]
+				local pos = spawn.pos
+
+				-- the scPos has to be calculatet inside a non modified cam3D space
+				-- to yield correct results
+				camStart3D()
+					scPos = pos:ToScreen()
+				camEnd3D()
+
+				local isHighlighted, dist3d = IsHighlighted(pos, scPos)
+
+				if util.IsOffScreen(scPos) or dist3d > maxEditDistance then continue end
+
+				if not isHighlighted then
+					renderDrawSphere(pos, 10, 4 + 750 / dist3d, 4 + 750 / dist3d, color)
+				else
+					proximitySpawns[#proximitySpawns + 1] = {
+						entType = entType,
+						spawnType = spawnType,
+						spawn = spawn,
+						dist3d = dist3d,
+						scPos = scPos,
+						id = i
+					}
+				end
+			end
+		end
+	end
+
+	local function RenderHook()
+		entspawnscript.SetFocusedSpawn(nil)
+
+		local spawnEntList = entspawnscript.GetSpawns()
+
+		if not spawnEntList then return end
+
+		local weps = spawnEntList[SPAWN_TYPE_WEAPON] or {}
+		local ammo = spawnEntList[SPAWN_TYPE_AMMO] or {}
+		local plys = spawnEntList[SPAWN_TYPE_PLAYER] or {}
+
+		local proximitySpawns = {}
+
+		renderSetColorMaterial()
+
+		local colorWeapon = ColorAlpha(entspawnscript.GetColorFromSpawnType(SPAWN_TYPE_WEAPON), 100)
+		local colorAmmo = ColorAlpha(entspawnscript.GetColorFromSpawnType(SPAWN_TYPE_AMMO), 100)
+		local colorPlayer = ColorAlpha(entspawnscript.GetColorFromSpawnType(SPAWN_TYPE_PLAYER), 100)
+
+		-- draw spawn bubbles and put bubbles that might be highlighted in
+		-- the 'proximitySpawns' table
+		PaintSpawns(SPAWN_TYPE_WEAPON, weps, colorWeapon, proximitySpawns)
+		PaintSpawns(SPAWN_TYPE_AMMO, ammo, colorAmmo, proximitySpawns)
+		PaintSpawns(SPAWN_TYPE_PLAYER, plys, colorPlayer, proximitySpawns)
+
+		-- sort the proximity spawns by distance
+		table.sort(proximitySpawns, function(a, b)
+			return a.dist3d < b.dist3d
+		end)
+
+		-- draw the proximity spawns and highlight teh first one
+		for i = 1, #proximitySpawns do
+			local proximitySpawn = proximitySpawns[i]
+			local spawn = proximitySpawn.spawn
+			local spawnType = proximitySpawn.spawnType
+			local pos = proximitySpawn.spawn.pos
+			local entType = proximitySpawn.entType
+			local id = proximitySpawn.id
+			local dist3d = proximitySpawn.dist3d
+			local color
+
+			scPos = proximitySpawn.scPos
+
+			if spawnType == SPAWN_TYPE_WEAPON then
+				color = colorWeapon
+			elseif spawnType == SPAWN_TYPE_AMMO then
+				color = colorAmmo
+			elseif spawnType == SPAWN_TYPE_PLAYER then
+				color = colorPlayer
+			end
+
+			if i == 1 then
+				local client = LocalPlayer()
+
+				-- make sure there is nothingin the way
+				local trace = util.TraceLine({
+					start = client:EyePos(),
+					endpos = pos,
+					filter = {client},
+					mask = MASK_SOLID
+				})
+
+				if not trace.HitWorld then
+					entspawnscript.SetFocusedSpawn(spawnType, entType, id, spawn)
+				end
+
+				renderDrawSphere(pos, sphereRadius * (trace.HitWorld and 1 or 1.1), 4 + 750 / dist3d, 4 + 750 / dist3d, ColorAlpha(color, 245))
```

Comment:
This means, that if you can see a part of a sphere which is behind a wall, it gets highlighted, but not set as focused spawn.
So either check another proximityspawn or check the actual position you are looking at of the "3D-Sphere" or dont highlight it, when it cant be edited.
![grafik](https://user-images.githubusercontent.com/72046466/131798180-c65be45f-790f-4e20-b4d3-76cc6132246a.png)

Reply:
I have to test this but I'm pretty sure your described case did not happen in the real world. Hm.

Reply:
![grafik](https://user-images.githubusercontent.com/72046466/131970040-aae0fc13-c1f8-4094-abd7-4195ab1aa745.png)